### PR TITLE
feat: cleaned up Shopping Cart component, Product Detail, and Checkout

### PR DESCRIPTION
**Pull request by gustavop-dev**
#### Overview

This Pull Request addresses a comprehensive cleanup and refactoring of three major components in the project: the Shopping Cart, Product Detail, and Checkout components. The goal was to improve code readability, maintainability, and performance.

#### Changes Made

1. **Shopping Cart Component:**
   - Removed unnecessary state variables and streamlined the state management.
   - Refactored component lifecycle methods to reduce redundancy.
   - Applied design for desktop version

2. **Product Detail Component:**
   - Applied design for desktop version

3. **Checkout Component:**
   - Applied design for desktop version

### CODE REVIEW

File: frontend/src/components/layouts/Header.vue [hunk 1]
```diff
@@ -28,13 +28,13 @@
           <img src="@/assets/images/logo.png" class="w-auto h-14">
         </RouterLink>
         <div class="hidden lg:flex lg:flex-1 justify-end">
-          <div class="flex cursor-pointer">
-            <ShoppingBagIcon class="size-5 text-black m-2" @click="toggleShoppingCart" />
-            <span v-if="totalCartProducts > 0" class="bg-gray-300 text-white rounded-full text-xs size-5 flex items-center justify-center shadow-lg m-2 ml-0">
-              ({{ totalCartProducts }})
+          <div class="relative cursor-pointer">
+            <ShoppingBagIcon class="size-5 text-black m-2" @click="showShoppingCart = true" />
+            <span @click="showShoppingCart = true" v-if="totalCartProducts > 0" class="absolute top-0 left-1/2 bg-primary_p text-white rounded-full text-xs size-5 flex items-center justify-center shadow-lg m-2 ml-0">
```
Comment (carlos18bp) [inline frontend/src/components/layouts/Header.vue:33]: Split this line

File: frontend/src/views/product/Detail.vue [hunk 2]
```diff
@@ -342,7 +350,7 @@
      * Decrease the product quantity by 1.
      */
     const decrementQuantity = () => {
-        if (productQuantity.value > 0) {
+        if (productQuantity.value > 1) {
```
Comment (carlos18bp) [inline frontend/src/views/product/Detail.vue:353]: Do a double check here, I think if you set 1, the user cannt remove the product in his shopping cart

Comment (gustavop-dev) [reply frontend/src/views/product/Detail.vue:358]: I double checked it, and the user can delete or remove the product without problems from his shopping cart.

File: frontend/src/components/layouts/Header.vue (hunk 1 above)
Comment (gustavop-dev) [reply frontend/src/components/layouts/Header.vue:33]: Done

**Review by carlos18bp — APPROVED** [review]

**Pull request merged**